### PR TITLE
svirt: support boot from image

### PR DESCRIPTION
**Pull request by mnowaksuse**
Bare bone verification run: http://assam.suse.cz/tests/4047:

```
  'args' => [
              {
                'size' => '20G',
                'file' => '/var/lib/openqa/share/factory/hdd/SLES-12-SP2-x86_64-Build2184-gnome.qcow2'
              }
            ],
  'console' => 'svirt',
  'function' => 'add_disk'
...
13:23:04.4437 484 Command executed: virsh  define /var/lib/libvirt/images/openQA-SUT-10.xml
13:23:04.5966 484 Command's stdout:
Domain openQA-SUT-10 defined from /var/lib/libvirt/images/openQA-SUT-10.xml


13:23:04.7354 484 Command executed: virsh  start openQA-SUT-10
13:23:05.4843 484 Command's stdout:
Domain openQA-SUT-10 started
```

### CODE REVIEW

**Comment by mnowaksuse** [comment]
Requires: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/237.

**Comment by okurz** [comment]
My general feeling is some more changes should go into the back end here

File: tests/shutdown/svirt_upload_assets.pm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+sub extract_assets {
+    my ($args) = @_;
+
+    my $name   = $args->{name};
+    my $format = $args->{format};
+
+    type_string("clear\n");
+    my $svirt_img_name = "/var/lib/libvirt/images/openQA-SUT-" . get_var('VIRSH_INSTANCE', 1) . ".img";
```
Comment (coolo) [inline tests/shutdown/svirt_upload_assets.pm]: please don't hardcode the name more often. The svirt has a name attribute

**Comment by coolo** [comment]
The problem with having this in the backend is that would make it generic, but it's not. This is highly specific to SUSE infrastructure. So it's perfectly fine here.

**Pull request merged**